### DISCUSSION TITLE
How can I create settings for a plugin?

### DISCUSSION THREAD

**Discussion by gabrielcarloto** [Flow-Launcher/Flow.Launcher#1297] [Q&A] [answered]
I am thinking of creating a plugin with nodejs and I am wondering how to create an input in the plugin settings, like this one:

![image](https://user-images.githubusercontent.com/68694907/180838198-7b7274e3-18a2-4369-a396-ed16e789b8ef.png)

I could be wrong, but I couldn't find anything about this in the docs - the only thing I found was this #657. I know that this plugin has a `SettingsTemplate.yml`, but just adding this does nothing.

**Comment by taooceros** [comment, marked as answer]
If you are writing a c# plugin, you will need to draw a usercontrol instead of using the SettingsTemplate.yml. Maybe we may extend that for c# plugin in the future, but for now you have to write a usercontrol in wpf.

Check this interface https://www.flowlauncher.com/docs/#/API-Reference/Flow.Launcher.Plugin/ISettingProvider

**Reply by gabrielcarloto** [reply]
I'm writing a js plugin actually.

**Reply by taooceros** [reply]
You can check the source code of the github launcher plugin. It shall has a settings template file. 
Maybe we shall update the docs for that Though.

**Reply by taooceros** [reply]
https://github.com/Garulf/Github-Quick-Launcher/blob/main/SettingsTemplate.yaml

The name is hardcoded, so only `SettingsTemplate.yaml` will work but not `SettingsTemplate.yml`

**Reply by gabrielcarloto** [reply]
Oh, sorry, `.yaml` works. I didn't notice that I was writing it wrong. Now, how do I access the values input by the user? Is it in `process.env.setting` or or something similar?

And yeah it would be nice to have this in the docs.

**Reply by taooceros** [reply]
> Oh, sorry, `.yaml` works. I didn't notice that I was writing it wrong. Now, how do I access the values input by the user? Is it in `process.env.setting` or or something similar?
> 
> 
> 
> And yeah it would be nice to have this in the docs.

It is within the json text sending in along the query.
I think you may be able to see it, but i am not familier with our JavaScript plugin structures.
